example/models.py

```diff
@@ -1,17 +1,37 @@
 from django import forms
 from django.db import models
 from modelcluster.fields import ParentalManyToManyField
-from wagtail.admin.edit_handlers import FieldPanel, StreamFieldPanel
-from wagtail.core.fields import StreamField
-from wagtail.core.models import Page
+from wagtail import VERSION as WAGTAIL_VERSION
+try:
+    from wagtail.admin.panels import FieldPanel
+except ImportError:
+    from wagtail.admin.edit_handlers import FieldPanel, StreamFieldPanel
+
+try:
+    from wagtail.fields import StreamField
+except ImportError:
+    from wagtail.core.fields import StreamField
+
+try:
+    from wagtail.models import Page
+except ImportError:
+    from wagtail.core.models import Page
+
 from wagtail.snippets.models import register_snippet
 from wagtail_wordpress_import.blocks import WPImportStreamBlocks
 from wagtail_wordpress_import.models import WPImportedPageMixin
 
 
 class TestPage(WPImportedPageMixin, Page):
-    body = StreamField(WPImportStreamBlocks)
+    
+    streamfield_kwargs = {"use_json_field": True} if WAGTAIL_VERSION >= (3, 0) else {}
+    body = StreamField(WPImportStreamBlocks, **streamfield_kwargs)
+    
     categories = ParentalManyToManyField("example.Category", blank=True)
+
+    try: StreamFieldPanel
+    except NameError: StreamFieldPanel = FieldPanel
```

Comment:
**Comment**: I don't think I've ever seen a try/except statement like this before. Interesting! Why not just do this within the earlier try/except statement:

```python
try:
    from wagtail.admin.panels import FieldPanel
    StreamFieldPanel = FieldPanel  # or within an else: block
except ImportError:
    from wagtail.admin.edit_handlers import FieldPanel, StreamFieldPanel
```

Or would the linters reject this?

Reply:
Thanks Nick: Fixed

I've opened a new issue: #160 to do some more work here in the future